### PR TITLE
Rename and document dataLoader

### PR DESCRIPTION
**Pull request by Dschoordsch**
Document some dataLoader classes and relationships. Renamed the main
dataLoader as it doesn't just load rethink entitities and prefixed
rethink loader with rethink.

### CODE REVIEW

**Comment by Dschoordsch** [comment]
@BartoszJarocki could you take a look? The failed build is because of master being broken.

File: packages/server/dataloader/LocalCache.ts [hunk 1]
```diff
@@ -5,6 +5,10 @@ import {RWrite, Updater} from './RethinkDBCache'
 const resolvedPromise = Promise.resolve()
 
 type Thunk = () => void
+
+/**
+ * This cache is only used for the User table, {@see ProxiedCache}, {@see ../db}
```
Comment (BartoszJarocki) [inline packages/server/dataloader/LocalCache.ts:10]: This is great and I'd personally love to see more [JSDocs](https://jsdoc.app/) in our codebase ❤️

File: packages/server/graphql/getDataLoader.ts [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-const getDataLoader = (did?: string, reuseableId?: string) => {
+/**
+ * A user can keep reusing their dataloader for 500ms until a mutation comes in.
+ * Once the user triggers a mutation, then the dataloader is no longer safe to reuse.
+ */
+const getDataLoader = (did?: string) => {
   // if a did was provided, attempt to reuse that, since a mutation likely filled it up before sharing it
   // if the viewer is logged in, give them their own dataloader that they can quickly reuse for subsequent requests or share with others
   // if not logged in, just make a new anonymous laoder.
-  const id = did || `${NODE_ID}:${reuseableId || numToBase64(nextId++)}`
```
Comment (mattkrick) [inline packages/server/graphql/getDataLoader.ts:16]: nice getting rid of that old reuseableId!

File: packages/server/dataloader/RootDataLoader.ts [hunk 3]
```diff
@@ -25,19 +26,19 @@ const loaderMakers = {
...
+type ForeignLoaderMakers = typeof rethinkForeignKeyLoaderMakers
 type ForeignLoaders = keyof ForeignLoaderMakers
-type Unforeign<T> = T extends LoaderMakerForeign<infer U> ? U : never
+type Unforeign<T> = T extends RethinkForeignKeyLoaderMaker<infer U> ? U : never
 type TypeFromForeign<T extends ForeignLoaders> = TypeFromPrimary<Unforeign<ForeignLoaderMakers[T]>>
 
 type CustomLoaderMakers = typeof customLoaderMakers & typeof atlassianLoaders & typeof pollLoaders
 type CustomLoaders = keyof CustomLoaderMakers
-type Uncustom<T> = T extends (parent: RethinkDataLoader) => infer U ? U : never
+type Uncustom<T> = T extends (parent: RootDataLoader) => infer U ? U : never
```
Comment (mattkrick) [inline packages/server/dataloader/RootDataLoader.ts:41]: oh gosh, i'm so sorry you had to look at this file. it's terrible.

**Review by mattkrick — APPROVED** [review]

**Comment by mattkrick** [comment]
thank you for taking this on, it's been on my mind ever since we started working more in PG.
ready to merge after fixing the conflict!

**Pull request merged**